File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    case listen(input) do
      :silence ->  "Fine. Be that way!"
      :question -> "Sure."
      :shouting -> "Whoa, chill out!"
      :whatever -> "Whatever."
    end
  end

  defp listen(input) do
    cond do
      String.trim(input) == "" ->
        :silence
      String.ends_with?(input, "?") ->
        :question
      String.upcase(input) == input and String.downcase(input) != input ->
        :shouting
      true -> :whatever
    end
  end
end

File: python/bob/bob.py
def hey(string):
    string = string.strip()
    if string == '':
        return 'Fine. Be that way!'
    elif string.isupper():
        return 'Whoa, chill out!'
    elif string.endswith('?'):
        return 'Sure.'
    else:
        return 'Whatever.'
